packages/cli/src/runners/__tests__/task-runner-server.test.ts

```diff
@@ -0,0 +1,45 @@
+import type { GlobalConfig } from '@n8n/config';
+import { mock } from 'jest-mock-extended';
+import { ServerResponse } from 'node:http';
+import type WebSocket from 'ws';
+
+import type { TaskRunnerAuthController } from '@/runners/auth/task-runner-auth.controller';
+import { TaskRunnerServer } from '@/runners/task-runner-server';
+
+import type { TaskRunnerServerInitRequest } from '../runner-types';
+
+describe('TaskRunnerServer', () => {
+	describe('handleUpgradeRequest', () => {
+		it('should close WebSocket when response status code is > 200', () => {
+			const ws = mock<WebSocket>();
+			const request = mock<TaskRunnerServerInitRequest>({
+				url: '/runners/_ws',
+				ws,
+			});
+
+			const server = new TaskRunnerServer(
+				mock(),
+				mock<GlobalConfig>({ taskRunners: { path: '/runners' } }),
+				mock<TaskRunnerAuthController>(),
+				mock(),
+			);
+
+			// @ts-expect-error Private property
+			server.handleUpgradeRequest(request, mock(), Buffer.from(''));
+
+			const response = new ServerResponse(request);
+			response.writeHead = (statusCode) => {
+				if (statusCode > 200) ws.close();
+				return response;
+			};
+
+			response.writeHead(401);
+			expect(ws.close).toHaveBeenCalledWith(); // no args
+
+			jest.clearAllMocks();
+
+			response.writeHead(200);
+			expect(ws.close).not.toHaveBeenCalled();
```

Comment:
nit: This could be it's own test case as the test case name implies we are testing only status code >200

Reply:
[726639a](https://github.com/n8n-io/n8n/pull/11633/commits/726639ac33fecafb3c2a5278c8999d0ab06507a7)